src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -301,12 +331,362 @@ TEST_F(RPCServiceImplTest,
+  SetHMIAPI(derived_api);
+  CSmartSchema schema;
+  hmi_so_factory_.GetSchema(function_id, message_type, schema);
+  auto initial_schema_item = schema.getSchemaItem();
+  // Change state of schema and add custom parameter
+  auto new_function_id = hmi_apis::FunctionID::Buttons_OnButtonPress;
+  std::map<std::string, SMember> members;
+  members["custom member"] = SMember();
+  rpc_service_->UpdateHMIRPCParams(function_id, message_type, members);
+  hmi_so_factory_.GetSchema(new_function_id, message_type, schema);
+  // Check that schema is changed
+
+  EXPECT_NE(initial_schema_item, schema.getSchemaItem());
+}
+
+TEST_F(RPCServiceImplTest, SendMessageToHMI_InvalidMessageType) {
+  auto message = CreateMessage(smart_objects::SmartType_Invalid);
+  (*message)[am::strings::msg_params][am::strings::function_id] = kFunctionId;
+  // hmi_apis::FunctionID::BasicCommunication_OnSystemCapabilityUpdated;
+
+  utils::Optional<am::plugin_manager::RPCPlugin> mock_rpc_plugin_opt(
+      mock_rpc_plugin_);
+  EXPECT_CALL(mock_app_mngr_, GetPluginManager())
+      .WillOnce(ReturnRef(mock_rpc_plugin_manager_));
+  EXPECT_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(_, _))
+      .WillOnce(Return(mock_rpc_plugin_opt));
+  EXPECT_CALL(mock_rpc_plugin_, PluginName()).WillOnce(Return(""));
+  rpc_service_->SendMessageToHMI(message);
+}
+
+TEST_F(RPCServiceImplTest, SendMessageToHMI_NoHMIHandler) {
+  auto message = CreateMessage(smart_objects::SmartType_Invalid);
+  PrepareBasicMessage(message);
+  (*message)[am::strings::msg_params][am::strings::result_code] =
+      ns_smart_device_link_rpc::V1::Result::SUCCESS;
+  rpc_service_->set_hmi_message_handler(nullptr);
+  EXPECT_CALL(mock_app_mngr_, GetPluginManager()).Times(0);
+}
+
+TEST_F(RPCServiceImplTest, Handle_NoHMIHandler) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  message_handler->set_hmi_message_handler(nullptr);
+  std::shared_ptr<am::Message> message_to_send(
+      new am::Message(protocol_handler::MessagePriority::kDefault));
+  EXPECT_CALL(mock_hmi_handler_, SendMessageToHMI(_)).Times(0);
+  message_handler->Handle(am::rpc_service::impl::MessageToHmi(message_to_send));
+}
+
+TEST_F(RPCServiceImplTest, Handle_ValidHMIHandler) {
+  auto message_handler = std::make_shared<rpc_service::RPCServiceImpl>(
+      mock_app_mngr_,
+      request_controller_,
+      &mock_protocol_handler_,
+      &mock_hmi_handler_,
+      mock_command_holder_,
+      mock_rpc_protection_manager_,
+      hmi_so_factory_,
+      mobile_so_factory_);
+  std::shared_ptr<am::Message> message_to_send(
+      new am::Message(protocol_handler::MessagePriority::kDefault));
+  EXPECT_CALL(mock_hmi_handler_, SendMessageToHMI(_)).Times(1);
```

Comment:
check SendMessageToHMI argument

Reply:
Times 1 is redundant. it is default dehaviour of expectation

Reply:
Fixed in commit 4b57f0d5561b3b6e822bdd865382187875b7a7aa